datasets/fr/amf_regulatory_sanctions/crawler.py

```diff
@@ -0,0 +1,161 @@
+from typing import Any, Generator, Optional, Tuple, Union
+
+from normality import collapse_spaces
+from zavod import Context, helpers as h
+import re
+from html import unescape
+from urllib.parse import urljoin
+from lxml import html
+
+
+CLEAN_ENTITY = re.compile(r"(<br />\r\n| et |;)", re.IGNORECASE)
+BASE_URL = "https://www.amf-france.org"
+
+
+def parse_json(context: Context) -> Generator[dict, None, None]:
+    response = context.fetch_json(context.data_url)
+    if "data" not in response:
+        context.log.info("No data available.")
+        return
+
+    context.log.info(
+        f"Fetched {len(response['data'])} results."
+    )  # Log the number of results
+    for item in response["data"]:
+        yield item
+
+
+def get_value(
+    data: dict, keys: Tuple[str, ...], default: Optional[Any] = None
+) -> Union[Optional[Any], Any]:
+    for key in keys:
+        val = data.pop(key, None)
+        if val is not None:
+            return val
+    return default
+
+
+PREFIXES = [
+    "Sociétés",
+    "Société",
+    "société",
+    "Banque",
+    "la société",
+    "Caisse",
+    "cabinet",
+    "La sociétés",
+    "les CABINETS",
+    "les sociétés",
+    "LA",
+    "de gestion",
+    "cabinets",
+    "SCS",  # Société en commandite simple
+    r"M\.?",
+    "Mme",
+    r"MM\.?",
+    "Madame",
+    "Monsieur",
+    "Monsieur",
+    "Melle",
+]
+
+
+comma_prefix_anon = "(,|et) +(" + "|".join(PREFIXES) + r")[\. \b]+[A-Z]\b"
+prefix_anon = "(" + "|".join(PREFIXES) + r")[\. \b]+[A-Z]\b"
+comma_anon = r"(,|et) +[A-Z]\b"
+
+
+def clean_names_str(names_str: str) -> Optional[str]:
+    names_str = re.sub(comma_prefix_anon, "", names_str)
+    names_str = re.sub(prefix_anon, "", names_str)
+    names_str = re.sub(comma_anon, "", names_str)
+    names_str = collapse_spaces(names_str)
+    return names_str
+
+
+def process_entity(
```

Comment:
another little nitpick, but mainly to help you see intent in naming in the codebase.

we would tend to call this function "crawl_entity" (search the codebase for crawl_ - there are a few :)

a name like "process_entity" is a bit like calling a variable "data". we know a variable holds data. What would be the most helpful name to help someone know what data is in it? similarly, we know a function processes stuff. At least we know this is working on an entity, but "crawl entity", at least in this codebase, helps us make a reasonable assumption that we're turning their data format into out data format and emitting the entity.

Reply:
nice one including the date in there